### PR TITLE
CBL-3788: Collection SG test, "Pull deltas from SG"

### PR DESCRIPTION
**Pull request by velicuvlad**

### CODE REVIEW

File: Replicator/tests/ReplicatorCollectionSGTest.cc [hunk 1]
```diff
@@ -697,6 +698,161 @@ TEST_CASE_METHOD(ReplicatorCollectionSGTest, "Multiple Collections Incremental R
...
 }
 
+TEST_CASE_METHOD(ReplicatorCollectionSGTest, "Pull deltas from Collection SG", "[.SyncServerCollection]") {
+    _authHeader = TestUserAuthHeader;
+    const string idPrefix = timePrefix();
+    // one collection now now. Will use multiple collection when SG is ready.
+    constexpr size_t collectionCount = 1;
+
+    constexpr size_t kDocBufSize = 60;
+    constexpr int kNumDocs = 50, kNumProps = 50;
```
Comment (jianminzhao) [inline Replicator/tests/ReplicatorCollectionSGTest.cc:708]: please put comment why reduce the number from 1000

Comment (velicuvlad) [reply Replicator/tests/ReplicatorCollectionSGTest.cc:708]: will add comment and once I discuss with SGW, will create ticket if necessary

File: Replicator/tests/ReplicatorCollectionSGTest.cc [hunk 2]
```diff
@@ -697,6 +698,161 @@ TEST_CASE_METHOD(ReplicatorCollectionSGTest, "Multiple Collections Incremental R
...
+    double timeWithDelta = 0, timeWithoutDelta = 0;
+    for (int pass = 1; pass <= 3; ++pass) {
+        if (pass == 3) {
+            C4Log("-------- DISABLING DELTA SYNC --------");
+            Encoder encDelta;
+            encDelta.beginDict();
+            encDelta.writeKey(C4STR(kC4ReplicatorOptionDisableDeltas));
+            encDelta.writeBool(true);
+            encDelta.endDict();
+            encDelta.finish();
```
[the remark this answers is not in the document]
Comment (jianminzhao) [reply Replicator/tests/ReplicatorCollectionSGTest.cc]: no, not remove, but to use it. please compare with the original test.

[the remark this answers is not in the document]
Comment (velicuvlad) [reply Replicator/tests/ReplicatorCollectionSGTest.cc]: `_options = AllocedDict(enc.finish());` - it is not used anywhere inside the original test

**Review by jianminzhao — APPROVED** [review]

**Pull request merged**